### PR TITLE
Add availability gating for async function and actor type definitions.

### PR DESCRIPTION
**Pull request by DougGregor**
Implements rdar://80098709.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test and merge

File: test/Concurrency/concurrency_availability.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// RUN: %target-swift-frontend -parse-stdlib -target x86_64-apple-macosx10.15 -typecheck -verify %s
+// RUN: %target-swift-frontend -parse-stdlib -target x86_64-apple-macosx12 -typecheck %s
+// REQUIRES: OS=macosx
+
+func f() async { } // expected-error{{concurrency is only available in}}
```
Comment (kavon) [inline test/Concurrency/concurrency_availability.swift:5]: nice 👍🏼

**Review by kavon — APPROVED** [review]

**Comment by DougGregor** [comment]
@swift-ci please test macOS

https://github.com/apple/llvm-project/pull/3115

**Comment by DougGregor** [comment]
https://github.com/apple/llvm-project/pull/3115

@swift-ci please test macOS

**Comment by DougGregor** [comment]
@swift-ci please test macOS

apple/llvm-project#3115

**Comment by DougGregor** [comment]
@swift-ci please test macOS

apple/llvm-project#3115

**Comment by DougGregor** [comment]
https://github.com/apple/llvm-project/pull/3115

@swift-ci please test macOS

**Comment by DougGregor** [comment]
apple/llvm-project#3115

@swift-ci please test macOS

**Pull request merged**